pkg/sentry/fsimpl/gofer/gofer.go

```diff
@@ -95,7 +95,42 @@ const (
 	cacheRemoteRevalidating  = "remote_revalidating"
 )
 
-const defaultMaxCachedDentries = 1000
+const (
+	defaultMaxCachedDentries         = 1000
+	defaultMaxCachedNegativeChildren = 1000
+)
+
+// +stateify savable
+type negativeNamesCache struct {
+	// namesList stores negative names with fifo list.
+	// name stored in namesList only means it used to be negative
+	// at the moment you pushed it to the list.
+	namesList stringList
+	namesLen  uint64
+}
+
+func (cache *negativeNamesCache) Empty() bool {
+	return cache.namesLen == 0
+}
+
+func (cache *negativeNamesCache) Init(len uint64) {
+	elements := make([]stringListElem, len)
+	for i := uint64(0); i < len; i++ {
+		cache.namesList.PushFront(&elements[i])
+	}
+	cache.namesLen = len
+}
+
+// Update will push name to the front of the list,
+// and pop the tail value.
+func (cache *negativeNamesCache) Update(name string) string {
```

Comment:
I think `Add` would be a better method name here. Since it is a fixed-size cache, the removal is implied. `Update` is sort of vague about what it is updating.

Reply:
done